ui/src/components/Leap/useLeap.ts

```diff
@@ -0,0 +1,261 @@
+            } else if (app === 'Groups' && o.title === 'Create New Group') {
+              modalNavigate(`/groups/new`, {
+                state: { backgroundLocation: location },
+              });
+            } else if (app === 'Talk' && o.title === 'Groups') {
+              window.open(
+                `${window.location.origin}/apps/groups/find`,
+                '_blank'
+              );
+            } else {
+              navigate(o.to);
+            }
+            setIsOpen(false);
+            setSelectedIndex(0);
+          },
+          resultIndex: idx,
+        }))
+      : [];
+
+  const shipResults = useMemo(() => {
+    if (inputValue === '') {
+      return [];
+    }
+
+    return [
+      {
+        section: 'Ships',
+      },
+      ...Object.entries(contacts)
+        .filter(
+          ([patp, contact]) =>
+            patp.toLowerCase().includes(inputValue.toLowerCase()) ||
+            contact.nickname.toLowerCase().includes(inputValue.toLowerCase())
+        )
+        .map(([patp, contact], idx) => {
+          const onSelect = () => {
+            if (app === 'Talk') {
+              navigate(`/dm/${patp}`);
+            } else {
+              modalNavigate(`/profile/${patp}`, {
+                state: { backgroundLocation: location },
+              });
+            }
+            setSelectedIndex(0);
+            setInputValue('');
+            setIsOpen(false);
+          };
+          return {
+            onSelect,
+            icon: PersonIcon,
+            input: inputValue,
+            title: contact.nickname
+              ? `${contact.nickname} (${cite(patp)})`
+              : cite(patp),
+            subtitle: (contact.status || contact.bio || '').slice(
+              0,
+              LEAP_DESCRIPTION_TRUNCATE_LENGTH
+            ),
+            to: `/profile/${patp}`,
+            resultIndex: idx,
+          };
+        }),
+    ];
+  }, [app, contacts, inputValue, location, modalNavigate, navigate]);
+
+  const channelResults = useMemo(() => {
+    if (inputValue === '') {
+      return [];
+    }
+
+    const allChannels = Object.entries(groups).reduce(
+      (memo, [groupFlag, group]) =>
+        [
+          ...memo,
+          Object.entries(group.channels).map(([nest, channel]) => ({
+            groupFlag,
+            group,
+            nest,
+            channel,
+          })),
+        ].flat(),
+      [] as {
+        groupFlag: string;
+        group: Group;
+        nest: string;
+        channel: GroupChannel;
+      }[]
+    );
+
+    return [
+      {
+        section: 'Channels',
+      },
+      ...allChannels
+        .filter(({ channel }) =>
+          channel.meta.title.toLowerCase().includes(inputValue.toLowerCase())
+        )
+        .map(({ groupFlag, group, channel, nest }, idx) => {
+          const [chType, chFlag] = nestToFlag(nest);
+          const onSelect = () => {
+            navigate(`/groups/${groupFlag}/channels/${nest}`);
```

Comment:
this will not navigate to the right place if you're in talk and choose a diary/heap